astropy/utils/slicing.py

```diff
@@ -0,0 +1,96 @@
+import numbers
+
+__all__ = ["simplify_basic_index"]
+
+
+def simplify_basic_index(basic_index, *, shape):
```

Comment:
@mhvk - as our resident Numpy expert, do you know if anything like this exists already in Numpy?

Reply:
I looked but couldn't find anything...

One function that is useful (and essentially guaranteed to stay around since it is used everywhere in numpy) is `np.core.multiarray.normalize_axis_index` -- saves you all those test for negative values!

Though perhaps in this case, since you have slices already, one can use `slc.indices(len)`?

Aside: maybe just have this in `shapes.py` rather than a new module?

Reply:
Good points, I'll implement this!

Reply:
I've switched to using ``slc.indices`` which simplifies things a lot and takes care of the negative indices, and I've moved this to the existing ``shapes`` module.